packages/backend/server/src/plugins/oauth/providers/google.ts

```diff
@@ -33,7 +33,7 @@ export class GoogleOAuthProvider extends AutoRegisteredOAuthProvider {
   getAuthUrl(state: string) {
     return `https://accounts.google.com/o/oauth2/v2/auth?${this.url.stringify({
       client_id: this.config.clientId,
-      redirect_uri: this.url.link('/oauth/callback'),
+      redirect_uri: this.url.link('/api/oauth/redirect'),
```

Comment:
changing redirect_uri requires update the config in providers, better keep it

Reply:
already update the provider config

Reply:
there are selfhost users make the oauth work now. it's not easy to notify them updating the config as well. keep the old routes

Reply:
we have not handle `/oauth` route in ingress, and i not prefer redirect page at frontend
server side redirect can avoid most attacks supported by browser extensions, and will not cause the login process to be terminated due to front-end errors
and the current front-end routing needs to load most of the JS before it can execute the redirection logic, which is too slow

Reply:
<img width="1120" alt="image" src="https://github.com/user-attachments/assets/5ff20e3a-f3bb-4f92-b813-f2d41d240529">
according to statistics, there are less than 20 event using unofficial OAuth login sources
i believe the current unofficial oauth user is very small